src/editors/containers/VideoGallery/hooks.js

```diff
@@ -0,0 +1,194 @@
+import React from 'react';
+import * as module from './hooks';
+import messages from './messages';
+import {
+  filterKeys,
+  filterMessages,
+  sortKeys,
+  sortMessages,
+  sortFunctions,
+} from './utils';
+
+export const state = {
+  highlighted: (val) => React.useState(val),
+  searchString: (val) => React.useState(val),
+  showSelectVideoError: (val) => React.useState(val),
+  showSizeError: (val) => React.useState(val),
+  sortBy: (val) => React.useState(val),
+  filertBy: (val) => React.useState(val),
+  hideSelectedVideos: (val) => React.useState(val),
+};
```

Comment:
I'm not sure, I understand the reason for this. 

Hooks won't work outside of React components and the state won't be shared, so not sure what this is accomplishing?

Reply:
I followed this aproach used in other parts of the code like [VideoEditor hooks](https://github.com/openedx/frontend-lib-content-components/blob/d2d52a8aa58b2cd29f5efb291b28fe442115c3e9/src/editors/containers/VideoEditor/hooks.js#L9) and [EditorContainer hooks](https://github.com/openedx/frontend-lib-content-components/blob/d2d52a8aa58b2cd29f5efb291b28fe442115c3e9/src/editors/containers/EditorContainer/hooks.js#L19). What I understand is that the `useState` hook is used in such a way that they can be used in React components

Reply:
**Update**

I found this documentation on why we use this aproach:

https://github.com/openedx/frontend-lib-content-components/blob/d2d52a8aa58b2cd29f5efb291b28fe442115c3e9/docs/decisions/0005-internal-editor-testability-decisions.md#usestate-and-mockusestate